### PR TITLE
[Merged by Bors] - feat: `stab s ⊓ stab t ≤ stab (s ∩ t)`

### PR DESCRIPTION
**Pull request by YaelDillies**
and other similar lemmas

Moves:
* `mem_stabilizer_of_finite_iff_smul_le` -> `mem_stabilizer_set_iff_subset_smul_set`
* `mem_stabilizer_of_finite_iff_le_smul` -> `mem_stabilizer_set_iff_smul_set_subset`

From Kneser (LeanCamCombi)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Pointwise/Stabilizer.lean [hunk 1]
```diff
@@ -132,6 +198,28 @@ end Finset
...
+  exact mem_stabilizer_finset_iff_smul_finset_subset
+
+@[deprecated (since := "2024-11-25")]
+alias mem_stabilizer_of_finite_iff_smul_le := mem_stabilizer_iff_subset_smul_set
+
+@[deprecated (since := "2024-11-25")]
+alias mem_stabilizer_of_finite_iff_le_smul := mem_stabilizer_iff_smul_set_subset
+
 @[to_additive]
 lemma mem_stabilizer_set_iff_subset_smul_set {s : Set α} (hs : s.Finite) :
```
Comment (b-mehta) [inline Mathlib/Algebra/Pointwise/Stabilizer.lean:224]: It looks like this lemma is already in this file as `mem_stabilizer_set_iff_subset_smul_set`.

Comment (YaelDillies) [reply Mathlib/Algebra/Pointwise/Stabilizer.lean:204]: That's *why* I had this feeling of déjà vu 😆

File: Mathlib/Algebra/Pointwise/Stabilizer.lean [hunk 2]
```diff
@@ -57,8 +59,72 @@ lemma stabilizer_mul_self (s : Set G) : (stabilizer G s : Set G) * s = s := by
...
+
+@[to_additive]
+lemma stabilizer_inf_stabilizer_le_stabilizer_inter :
+    stabilizer G s ⊓ stabilizer G t ≤ stabilizer G (s ∩ t) := by
+  aesop (add simp [SetLike.le_def, smul_set_inter])
+
+@[to_additive]
+lemma stabilizer_inf_stabilizer_le_stabilizer_sdiff :
+    stabilizer G s ⊓ stabilizer G t ≤ stabilizer G (s \ t) := by
+  aesop (add simp [SetLike.le_def, smul_set_sdiff])
```
Comment (b-mehta) [inline Mathlib/Algebra/Pointwise/Stabilizer.lean:75]: Hmm, these three lemmas make it seems like there's some common generalisation here... If nothing else, can we derive one from the other two by saying something about compls?

File: Mathlib/Algebra/Pointwise/Stabilizer.lean [hunk 3]
```diff
@@ -148,6 +216,12 @@ lemma mem_stabilizer_set_iff_smul_set_subset {s : Set α} (hs : s.Finite) :
   rw [stabilizer_coe_finset, mem_stabilizer_finset_iff_smul_finset_subset, ← Finset.coe_smul_finset,
     Finset.coe_subset]
 
+@[deprecated (since := "2024-11-25")]
+alias mem_stabilizer_of_finite_iff_smul_le := mem_stabilizer_set_iff_subset_smul_set
+
+@[deprecated (since := "2024-11-25")]
+alias mem_stabilizer_of_finite_iff_le_smul := mem_stabilizer_set_iff_smul_set_subset
+
```
Comment (b-mehta) [inline Mathlib/Algebra/Pointwise/Stabilizer.lean:224]: Let's add these renames to the PR description, currently that only mentions adding new lemmas and not the (unrelated) change of deprecating these lemmas.

**Review by b-mehta — APPROVED** [review]
bors d+

File: Mathlib/Algebra/Pointwise/Stabilizer.lean (hunk 2 above)
Comment (YaelDillies) [reply Mathlib/Algebra/Pointwise/Stabilizer.lean:75]: The generalisation is that `stabilizer G s ⊓ stabilizer G t ≤ stabilizer G (f s t)` as long as `∀ a, a • f s t = f (a • s) (a • t)`

Comment (b-mehta) [reply Mathlib/Algebra/Pointwise/Stabilizer.lean:75]: oh neat, that's like a binary mul action hom

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**